BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -163,65 +169,103 @@ extension SBAProfileItem {
         
         switch self.itemType {
         case SBAProfileTypeIdentifier.string:
-            return newValue as? String != nil
+            return true // anything can be cast to a string
             
         case SBAProfileTypeIdentifier.number:
-            return newValue as? NSNumber != nil
+            if (newValue as? NSNumber != nil) {
+                return true
+            }
+            else if let quantity = newValue as? HKQuantity {
+                return quantity.is(compatibleWith: self.unit ?? commonDefaultUnit())
+            }
+            return false
             
         case SBAProfileTypeIdentifier.bool:
             return newValue as? NSNumber != nil
             
         case SBAProfileTypeIdentifier.date:
-            return newValue as? NSDate  != nil
+            return newValue as? NSDate != nil
             
         case SBAProfileTypeIdentifier.hkBiologicalSex:
             return newValue as? HKBiologicalSex != nil
             
         case SBAProfileTypeIdentifier.hkQuantity:
-            return newValue as? NSNumber != nil
+            guard let quantity = newValue as? HKQuantity else { return false }
+            return quantity.is(compatibleWith: self.unit ?? commonDefaultUnit())
             
         default:
-            return false
+            return true   // Any extended type isn't included in the common validation
+        }
+    }
+    
+    func commonDefaultUnit() -> HKUnit {
+        guard let option = SBAProfileInfoOption(rawValue: self.profileKey)
+        else {
+            return HKUnit.count()
+        }
+        switch option {
+        case .height:
+            return HKUnit(from: .centimeter)
+            
+        case .weight:
+            return HKUnit(from: .kilogram)
+            
+        default:
+            return HKUnit.count()
         }
     }
 }
 
 open class SBAProfileItemBase: NSObject, SBAProfileItem {
+    
     /**
      The value property is used to get and set the profile item's value in whatever internal data
      storage is used by the implementing class.
      */
-    open var value: Any?
+    open var value: Any? {
+        get {
+            // Look at the sourceKey, if not found then fall back to the fallback key and check that
+            let value = storedValue(forKey: sourceKey)
+            if value == nil, let fallback = fallbackKey {
+                return storedValue(forKey: fallback)
```

Comment:
yup, that's why I went with this. there are enough loops already b/c of the SBAUser. Also, I kind of figured that it doesn't totally make sense for the question type wouldn't change (and therefore where it is stored wouldn't change). You can still override the default implementation for either `storedValue()` or `value`.